### PR TITLE
[stable/mongodb-replicaset] feature: add support for extra containers and volumes

### PR DESCRIPTION
**Pull request by oba11**
#### Is this a new chart
No

#### What this PR does / why we need it:
Adds support for adding extra containers (as sidecar) and extra volumes (which can be used by the sidecar containers). This is useful for integrating backup sidecar e.g [mgod](https://github.com/stefanprodan/mgob) for backup/restore from external storage

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[stable/mychartname]`)

### CODE REVIEW

File: stable/mongodb-replicaset/Chart.yaml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 apiVersion: v1
 name: mongodb-replicaset
 home: https://github.com/mongodb/mongo
-version: 3.11.5
+version: 3.11.6
```
Comment (steven-sheehy) [inline stable/mongodb-replicaset/Chart.yaml]: Minor version increase for new features

File: stable/mongodb-replicaset/README.md [hunk 2]
```diff
@@ -55,6 +55,7 @@ The following table lists the configurable parameters of the mongodb chart and t
 | `securityContext.fsGroup`           | Group ID for the container                                                | `999`                                               |
 | `securityContext.runAsUser`         | User ID for the container                                                 | `999`                                               |
 | `securityContext.runAsNonRoot`      |                                                                           | `true`                                              |
+| `sidecarContainers`                 | Array of snippets with your sidecar containers for mongodb-replicasets    | `[]`                                                |
```
Comment (steven-sheehy) [inline stable/mongodb-replicaset/README.md]: I'd prefer `extraContainers`. The description should say something more like "Additional containers to add to the StatefulSet"

File: stable/mongodb-replicaset/templates/mongodb-statefulset.yaml [hunk 3]
```diff
@@ -293,6 +293,9 @@ spec:
             initialDelaySeconds: 30
             periodSeconds: 10
 {{ end }}
+        {{- if .Values.sidecarContainers }}
+        {{- toYaml .Values.sidecarContainers | nindent 8 }}
```
Comment (steven-sheehy) [inline stable/mongodb-replicaset/templates/mongodb-statefulset.yaml]: This should probably be ran through tpl first: `{{ tpl (toYaml .Values.extraContainers) . | nindent 8 }}`

File: stable/mongodb-replicaset/templates/mongodb-statefulset.yaml [hunk 4]
```diff
@@ -329,6 +332,9 @@ spec:
           emptyDir: {}
         - name: configdir
           emptyDir: {}
+{{- if .Values.extraVolumes }}
+{{ toYaml .Values.extraVolumes | indent 8}}
```
Comment (steven-sheehy) [inline stable/mongodb-replicaset/templates/mongodb-statefulset.yaml]: Indent over and use nindent like you do above

**Review by steven-sheehy — CHANGES_REQUESTED** [review]
/assign

**Comment by oba11** [comment]
Thanks @steven-sheehy for taking the time to review my pull request. I've updated the code to the recommended changes and good for another review. Thanks again

**Comment by oba11** [comment]
Hello @steven-sheehy, just want to follow up with reviewing this PR again. Thank you so much

File: stable/mongodb-replicaset/Chart.yaml [hunk 5]
```diff
@@ -1,7 +1,7 @@
 apiVersion: v1
 name: mongodb-replicaset
 home: https://github.com/mongodb/mongo
-version: 3.11.5
+version: 3.12.5
```
Comment (steven-sheehy) [inline stable/mongodb-replicaset/Chart.yaml]: 3.12.0

Comment (oba11) [reply stable/mongodb-replicaset/Chart.yaml]: @steven-sheehy I've updated the version too. Thanks

**Review by steven-sheehy — APPROVED** [review]
/lgtm

**Comment by oba11** [comment]
thank you so much @steven-sheehy

**Pull request merged**